src/services/ClientService.java

```diff
@@ -0,0 +1,119 @@
+
+        if (noAccount > accountList.size()) {
+            throw new AppException(appMessage.getNO_ACCOUNT_FOUND());
+        }
+        return accountList.get(noAccount).getAmount();
+    }
+
+    public boolean transferMoney(Client client, String iban, double sum) {
+        List<DebitBankAccountDTO> accountList = client.getDebitAccountList();
+
+        return this.accountService.processDebitAccountTransfer(accountList, iban, sum);
+    }
+
+    public boolean operationBetweenAccounts(Client client,
+                                            String accountIbanFrom,
+                                            String accountIbanTo,
+                                            double sum) throws AppException {
+
+        List<DebitBankAccountDTO> accountList = client.getDebitAccountList();
+        DebitBankAccountDTO accountFrom = this.accountService.returnDebitAccount(accountList, accountIbanFrom);
+        DebitBankAccountDTO accountTo = this.accountService.returnDebitAccount(accountList, accountIbanTo);
+
+
+        if (accountFrom == null || accountTo == null) {
+            return false;
+        }
+
+        if (accountFrom.getAmount() < sum) {
+            throw new AppException(this.appMessage.getINSUFFICIENT_FUNDS());
+        }
+
+        this.accountService.processAccountsTransfer(accountFrom, accountTo, sum);
+        return true;
+    }
+
+    public boolean addMoneyToCreditAccount(Client client, String id, double sum) {
+        List<CreditBankAccountDTO> accountList = client.getCreditAccountList();
+
+        return (this.accountService.processCreditAccountTransfer(accountList, id, sum));
+    }
+
+    public String withdrawFromDebitAccount(Client client, String iban, double sum) throws AppException {
+        List<DebitBankAccountDTO> debitList = client.getDebitAccountList();
+
+        for (DebitBankAccountDTO account : debitList) {
+            if (account.getIban().equals(iban)) {
+                if (account.getAmount() < sum) {
+                    throw new AppException(appMessage.getINSUFFICIENT_FUNDS());
+                }
+                this.accountService.processWithDraw(account, sum);
+                return this.appMessage.getSUCCESS_MSG();
+            }
+        }
+        throw new AppException(appMessage.getINCORRECT_IBAN());
+    }
+
+    public String createReport(String userName, String ownerId) {
+        List<Transaction> transactionList = this.accountService.getTransactionCache().getTransactionList();
+        List<Transaction> filteredList = this.filterTransactions(transactionList, ownerId);
+
+        String finalMsg = appMessage.getREPORT_MSG().replace("%CLIENT_USERNAME%", userName);
+        String tempMsg = "";
+
+        for(Transaction trans : filteredList) {
+            tempMsg += "Id: " + trans.getId() +
+                       " Type: " + trans.getTransType().toString() +"\n";
+        }
+
+        finalMsg += tempMsg;
+        return finalMsg;

```

Comment:
you can another field Transaction to store         String finalMsg = appMessage.getREPORT_MSG().replace("%CLIENT_USERNAME%", userName);
 then you can return transaction.toString();